lightning/src/ln/channel.rs

```diff
@@ -15985,7 +16008,7 @@ mod tests {
 			amount_msat: htlc_amount_msat,
 			payment_hash: PaymentHash(Sha256::hash(&[42; 32]).to_byte_array()),
 			cltv_expiry: 300000000,
-			state: InboundHTLCState::Committed,
+			state: InboundHTLCState::Committed { update_add_htlc_opt: None },
```

Comment:
Is it ok to never populate `update_add_htlc_opt` in the tests in this commit, if that isn't supposed to happen in prod?

Reply:
Yeah, because that field is only used by the manager on restart. Hopefully it's clearer with docs are on the field.

Reply:
I thought maybe the copying of the update_add message into InboundHTLCState needs to be covered in this commit. It isn't currently. But coverage is probably obtained later on.